operator/controllers/common/metrics.go

```diff
@@ -45,27 +45,45 @@ func ObserveDeploymentInterval(ctx context.Context, client client.Client, reconc
 		return err
 	}
 
-	for _, ro := range piWrapper.GetItems() {
+	items := piWrapper.GetItems()
+	for index := 0; index < len(items); index++ {
+		ro := items[index]
 		reconcileObject, _ := interfaces.NewMetricsObjectWrapperFromClientObject(ro)
 		if reconcileObject.GetPreviousVersion() != "" {
-			err := client.Get(ctx, types.NamespacedName{Name: fmt.Sprintf("%s-%s", reconcileObject.GetParentName(), reconcileObject.GetPreviousVersion()), Namespace: reconcileObject.GetNamespace()}, previousObject)
-			if err != nil {
-				return nil
-			}
-			piWrapper2, err := interfaces.NewMetricsObjectWrapperFromClientObject(previousObject)
-			if err != nil {
-				return err
+			if !reconcileObject.IsEndTimeSet() {
+				continue
 			}
-			if reconcileObject.IsEndTimeSet() {
-				previousInterval := reconcileObject.GetEndTime().Sub(piWrapper2.GetStartTime())
-				gauge.Observe(ctx, previousInterval.Seconds(), reconcileObject.GetDurationMetricsAttributes()...)
+			predecessor := getPredecessor(reconcileObject, items)
+			if predecessor == nil {
+				continue
 			}
+
+			previousInterval := reconcileObject.GetEndTime().Sub(predecessor.GetStartTime())
+			gauge.Observe(ctx, previousInterval.Seconds(), reconcileObject.GetDurationMetricsAttributes()...)
 		}
 	}
 
 	return nil
 }
 
+func getPredecessor(successor *interfaces.MetricsObjectWrapper, items []client.Object) interfaces.MetricsObject {
+	var predecessor interfaces.MetricsObject
+	for i := 0; i < len(items); i++ {
+		// to calculate the interval, we take the earliest revision of the previous version as a reference
+		if strings.HasPrefix(items[i].GetName(), fmt.Sprintf("%s-%s", successor.GetParentName(), successor.GetPreviousVersion())) {
+			predecessorCandidate, err := interfaces.NewMetricsObjectWrapperFromClientObject(items[i])
+			if err != nil {
+				// continue with the other items
+				continue
+			}
+			if predecessor == nil || predecessorCandidate.GetStartTime().Before(predecessor.GetStartTime()) {
+				predecessor = predecessorCandidate
```

Comment:
Previous items are not sorted by timestamp, so any revision could be used as a baseline for computing the previous interval. I think this is worth an issue/discussion on its own to see the consensus from the community on what should be the previous point to consider:
- any previous version item
- last revision of the previous app version